operator/pkg/helmreconciler/pruning.go

```diff
@@ -17,38 +17,32 @@ package helmreconciler
 import (
 	"context"
 
-	"istio.io/pkg/log"
-
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	utilerrors "k8s.io/apimachinery/pkg/util/errors"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"istio.io/istio/operator/pkg/object"
+	"istio.io/pkg/log"
 )
 
 // Prune removes any resources not specified in manifests generated by HelmReconciler h. If all is set to true, this
 // function prunes all resources.
-func (h *HelmReconciler) Prune(all bool) error {
+func (h *HelmReconciler) Prune(excluded map[string]bool, all bool) error {
 	allErrors := []error{}
-	namespacedResources, nonNamespacedResources := h.customizer.PruningDetails().GetResourceTypes()
+	namespacedResources, _ := h.customizer.PruningDetails().GetResourceTypes()
```

Comment:
Is there a reason you're leaving non-namespaced resources around?  We should be cleaning up things like `ClusterRole`, etc.

Reply:
It's to make rollback possible - at least we can't delete CRDs because that will delete user config. Can other the other cluster resources be deleted? I wasn't sure.

Reply:
If CRDs aren't in the type list, you don't have to worry about them being deleted.  If the cluster resource wasn't installed, I think it's safe to assume it's not needed.  If it breaks something, that's an issue with the rendered charts (e.g. deleting a ClusterRole that's used by a RoleBinding or ClusterRoleBinding, etc.).

Also, it's not relevant to this PR, but since you brought it up, CRD installation should be part of a general bootstrapping operation, rather than part of control plane installation.

Reply:
i've removed the CRD from the list and left everything else for removal.